model/base_constraints.py

```diff
@@ -65,7 +67,7 @@ def add_maximum_demand_coverage(self, mu):
             (
                 mu[c, t] <= self.demand["max"][c, t] - self.demand["min"][c, t]
                 for c in self.competencies
-                for t in self.time_periods
+                for t in self.time_periods[c]
```

Comment:
Is `self.time_periods` now a dict with competencies as keys and the relevant time periods as values?

Reply:
Yes